### ISSUE TITLE
Occasional CORS loading issue on domtoimage + Leaflet + Stamen (toner-lite)

### ISSUE THREAD

**Issue by MSCAU** [stamen/maps.stamen.com#158]
After having seen CORS errors for the last few days when loading Stamen tiles via [domtoimage](https://github.com/tsayen/dom-to-image) and [Leaflet](https://leafletjs.com/), inexplicably those errors had disappeared for most of today. They are now back (the last 30 mins). I can't imagine what else it could be. Are you people modifying the CORS settings at the moment?

**Comment by MSCAU** [comment]
FWIW, I am not getting CORS errors for the last 24 hours. Hooray.

**Comment by MSCAU** [comment]
Hmm. Am now getting these errors again for the last few weeks but only in **Chrome 86.0.4240.198 (Official Build) (64-bit)**, not in **Firefox 82.0.3 (64-bit)**.

To be clear, the tiles load initially (via Leaflet), but then fail to load in Chrome (and Edge) when requested a second time through domtoimage.

**Comment by MSCAU** [comment]
Somewhat surprised to find these errors have now disappeared in Chrome and Edge. Although I have just updated Chrome to the latest version 86.0.4280.66 (Windows 10), and initially thought that may have fixed it, I see that the problem has also disappeared from Edge, despite the fact that this hadn't been updated since last week.

Issue persists in Chrome 86.0.4280.67 (Mac) and Safari, though. It looked like a browser issue, but now it seems it may not be...

**Comment by anti-destin** [comment]
Just wanted to follow up on this, as we have been experiencing CORS errors also when trying to export a Leaflet map using html-to-image: https://github.com/bubkoo/html-to-image.

It used to work smoothly before, so it is not clear what has changed.

Currently using:
Chrome 101.0.4951.54
https://stamen-tiles-{s}.a.ssl.fastly.net/toner-lite/{z}/{x}/{y}{r}.png

@MSCAU @ptkz have you found any solutions?

[Screen Shot 2022-05-09 at 09 18 31](https://user-images.githubusercontent.com/11545842/167423144-3ca85d46-c20d-41e6-bf60-1849bce6fe12.png)

**Comment by MSCAU** [comment]
@anti-destin yes. We ended up setting up a proxy, ie. sending the requests to **fastly.net** from the server instead of the browser, and then serving the titles to the browser from the same domain as the rest of the page. This works well in all browsers and we haven't had a CORS warning or error since.

**Comment by anti-destin** [comment]
@MSCAU thanks for the insight! Very helpful.
